theano/compile/tests/test_function_module.py

```diff
@@ -383,6 +383,13 @@ def test_copy_delete_updates(self):
             assert cpy(1)[0] == 4
             assert cpy(1)[0] == 4
 
+        # Test if unused implicit inputs from delete_updates are ignored
+        # as intended.
+        for mode in ["FAST_RUN", "FAST_COMPILE"]:
+            ori = theano.function([x], x, mode=mode, updates={z: z * 2})
+            cpy = ori.copy(delete_updates=True)
+
```

Comment:
Does it make the first test redundant, or do they cover separate cases?

On Fri, Apr 22, 2016 at 8:03 AM, Frédéric Bastien <notifications@github.com>
wrote:

> In theano/compile/tests/test_function_module.py
> <https://github.com/Theano/Theano/pull/4419#discussion_r60728296>:
>
> > @@ -383,6 +383,13 @@ def test_copy_delete_updates(self):
> >              assert cpy(1)[0] == 4
> >              assert cpy(1)[0] == 4
> >
> > +        # Test if unused implicit inputs from delete_updates are ignored
> > +        # as intended.
> > +        for mode in ["FAST_RUN", "FAST_COMPILE"]:
> > +            ori = theano.function([x], x, mode=mode, updates={z: z * 2})
> > +            cpy = ori.copy(delete_updates=True)
> > +
>
> Can you add this extra test that make one explicit input useless
>
>              ori = theano.function([x, y], x, mode=mode, updates={z: z + y})
>              cpy = ori.copy(delete_updates=True)
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/Theano/Theano/pull/4419/files/b96e3954b09296326f768af7fc3eae85eb7b87ab..03281b5f61908a23c3cdb07f78f80607b307caa3#r60728296>
>